docs/enhancements/third-party-cni-support.md

```diff
@@ -0,0 +1,269 @@
+
+**Manifest Generation:**
+Modify install-config generation to skip OVN-K-specific configuration when a third-party CNI is selected. The CNI will be configured via custom manifests provided by the user through the existing manifests API.
+
+**Manifest Validation:**
+Basic validation will be performed on uploaded manifests to ensure they are valid YAML/JSON and contain Kubernetes resource definitions. However, CNI-specific semantic validation (e.g., verifying Cilium operator configuration) is out of scope. Invalid manifests will cause installation failures during cluster bootstrapping, which is acceptable as users are responsible for providing correct CNI configurations.
+
+**Feature Support Levels:**
+Update the feature support levels API to expose CNI support per platform and OCP version, enabling the UI to dynamically show or hide CNI options based on cluster configuration.
+
+#### Changes to assisted-installer-ui
+
+The Networking configuration step will be redesigned to support CNI selection:
+
+**Networking Step Layout:**
+- Cluster-Managed Networking option displays OVN-Kubernetes or OpenShiftSDN based on OCP version
+- User-Managed Networking option adds a CNI selection dropdown with the following options:
+  - OVN-Kubernetes (default for newer versions)
+  - OpenShiftSDN (available for older OCP versions)
+  - Cisco ACI
+  - Isovalent Cilium
+  - Tigera Calico
+  - None (Custom CNI)
+
+**Validation and Warnings:**
+- Real-time validation displays compatibility status:
+  - ✓ Success: "Cisco ACI is supported for OCP 4.18 on vSphere"
+  - ✗ Error: "Calico is not supported on AWS for OCP 4.15. Supported platforms: Bare Metal, vSphere, OpenStack"
+- Warning banner displayed when third-party CNI is selected:
+  - "Third-party CNIs require custom manifests to be uploaded. See CNI vendor documentation for manifest examples."
+
+**Manifest Requirements:**
+- **IMPORTANT**: Users selecting any CNI option other than OVN-Kubernetes or OpenShiftSDN (i.e., Cisco ACI, Cilium, Calico, or None) **must** upload CNI manifests via the Custom Manifests step. The installation will fail validation if third-party CNI is selected but no CNI-related manifests are provided.
+- UI will display a link to CNI vendor documentation for obtaining correct manifests
+- Manifest upload section will highlight required CNI operator and configuration manifests
+
+**Documentation Links:**
+Each CNI option displays a link to vendor-specific documentation for post-install configuration and manifest examples.
+
+#### Kubernetes API Changes
+
+Update `AgentClusterInstall` CR to support CNI selection through a new `networkType` field in the `spec.networking` section. Supported values include `OVNKubernetes`, `OpenShiftSDN`, `CiscoACI`, `Cilium`, `Calico`, and `None`.
```

Comment:
Is there an existing status field that can indicate whether or not the CNI validations are passing? 

We want the user to be able to see from the Kube API what's going on in case they specify one of these but fail to upload or provide the manifests, for example